File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;

        int output = Integer.MIN_VALUE;
        
        while (left < right) {
            int width = right - left;
            int lowHigh = 0;

            if (height[left] > height[right]) {
                lowHigh = height[right];
                right -= 1;
            } else {
                lowHigh = height[left];
                left += 1;
            }

            output = Math.max(output, width*lowHigh);
        }

        return output;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.js
var maxArea = function(height) {
    let max = 0;
    let start = 0;
    let end = height.length - 1;

    while (start < end) {
        let heightLeft = height[start];
        let heightRight = height[end];

        let hei = Math.min(heightLeft, heightRight);
        let wid = end - start;

        max = Math.max(max, hei * wid);

        if (heightRight > heightLeft) {
            start += 1;
        } else {
            end -= 1;
        }
    }

    return max;
};
